contentprovider.go

```diff
@@ -355,11 +399,13 @@ func (p *contentProvider) matchScore(secs []DocumentSection, m *LineMatch, langu
 		score.score += s
 	}
 
-	for _, f := range m.LineFragments {
-		startBoundary := f.LineOffset < len(m.Line) && (f.LineOffset == 0 || byteClass(m.Line[f.LineOffset-1]) != byteClass(m.Line[f.LineOffset]))
+	for _, r := range m.Ranges {
+		// calculate the start and end offset relative to the start of the content
+		relStartOffset := r.Start.ByteOffset - m.ContentStart.ByteOffset
+		relEndOffset := r.End.ByteOffset - m.ContentStart.ByteOffset
 
-		end := int(f.LineOffset) + f.MatchLength
-		endBoundary := end > 0 && (end == len(m.Line) || byteClass(m.Line[end-1]) != byteClass(m.Line[end]))
+		startBoundary := relStartOffset < len(m.Content) && (relStartOffset == 0 || byteClass(m.Content[relStartOffset-1]) != byteClass(m.Content[relStartOffset]))
+		endBoundary := relEndOffset > 0 && (relEndOffset == len(m.Content) || byteClass(m.Content[relEndOffset-1]) != byteClass(m.Content[relEndOffset]))
```

Comment:
I see we have this in the old code, but if I see End is documented as exclusive. doesn't this mean relEndOffset could be out of bounds?